tests/test_enterprise/api/test_filters.py

```diff
@@ -13,30 +13,63 @@
 
 from django.conf import settings
 
-from test_utils import TEST_EMAIL, TEST_PASSWORD, TEST_USERNAME, APITest, factories
+from test_utils import FAKE_UUIDS, TEST_EMAIL, TEST_PASSWORD, TEST_USERNAME, APITest, factories
 
 
 @ddt.ddt
 @mark.django_db
-class TestEnterpriseCustomerUserFilterBackend(APITest):
+class TestUserFilterBackend(APITest):
     """
-    Tests for enterprise API throttling.
+    Test suite for the ``UserFilterBackend`` filter.
     """
 
-    def create_user(self, username=TEST_USERNAME, password=TEST_PASSWORD, **kwargs):
+    def setUp(self):
+        self.create_user(id=1)
+        self.enterprise_customer = factories.EnterpriseCustomerFactory(uuid=FAKE_UUIDS[0])
+        self.enterprise_customer_user = factories.EnterpriseCustomerUserFactory(
+            enterprise_customer=self.enterprise_customer,
+            user_id=self.user.id,
+        )
+        self.client = APIClient()
+        self.client.login(username=TEST_USERNAME, password=TEST_PASSWORD)
+
+    @ddt.data(
+        (False, 1),
+        (True, 2),
+    )
+    @ddt.unpack
+    def test_filter_for_list(self, is_staff, expected_data_length):
         """
-        Create a test user and set its password.
+        Filter users based off of user ID -- show all for staff, but only the authenticated user himself otherwise.
         """
-        self.user = factories.UserFactory(username=username, is_active=True, **kwargs)
-        self.user.set_password(password)  # pylint: disable=no-member
-        self.user.save()  # pylint: disable=no-member
+        self.user.is_staff = is_staff
+        self.user.save()
+        factories.EnterpriseCustomerFactory()
+        response = self.client.get(settings.TEST_SERVER + reverse('enterprise-customer-list'))
+        assert response.status_code == status.HTTP_200_OK
+        data = self.load_json(response.content)
+        assert len(data['results']) == expected_data_length
 
-    def setUp(self):
-        """
-        Perform operations common for all tests.
+    @ddt.data(
+        (FAKE_UUIDS[0], {'uuid': FAKE_UUIDS[0], 'active': True}),
+        ('fake-uuid', {'detail': 'Not found.'})
+    )
+    @ddt.unpack
+    def test_filter_for_detail(self, uuid, expected_content_in_response):
```

Comment:
Ah I forgot the docstring here, thanks! :+1: 

I think you bring up a good point about distinguishing between staff/non-staff here. We should probably cover this based off of staff status rather than a valid UUID because we don't care about UUID validity here, we're just filtering.